### PR TITLE
run unit tests for Julia 1.6 until 1.9

### PR DESCRIPTION
**Pull request by SimeonEhrig**
- tests also the latest unreleased version 1.10 (can fail)
- integration tests still testing a single version. Will be extended in another PR.

### CODE REVIEW

**Comment by SimeonEhrig** [comment]
@szabo137 What do you think. The Unit tests passed successful version 1.6 until 1.9: https://gitlab.com/hzdr/qedjl-project/QEDbase-jl/-/pipelines/1032135149

Version 1.10 failed, but this is okay for the CI: https://gitlab.com/hzdr/qedjl-project/QEDbase-jl/-/jobs/5262079422

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -25,6 +24,18 @@ unit_tests_julia1.9:
...
+unit_tests_releases:
+  extends: .untit_test_template
+  parallel:
+    matrix:
+      - JULIA_VERSION: ["1.6", "1.7", "1.8", "1.9"]
+  image: julia:$JULIA_VERSION
+
+unit_tests_nightly:
+  extends: .untit_test_template
+  image: julia:1.10-rc
```
Comment (szabo137) [inline .gitlab-ci.yml:36]: wouldn't it be better to use the image `julia:nightly`, so there is less maintenance effort?

**Review by szabo137 — DISMISSED** [review]
I was a bit more lenient about the Julia versions which should be checked. Are those runs executed in parallel, or might this add time to the ci-runs?

**Comment by szabo137** [comment]
BTW: nightly fails are fine. These bugs will become a problem if nightly becomes stable, and this will be the time when we *need* to do something - before we *can* do it.

File: .gitlab-ci.yml (hunk 1 above)
Comment (SimeonEhrig) [reply .gitlab-ci.yml]: There is no `julia:nightly`. Now, I use the same baseimage like the official julia container and install the nightly version at the begin of each test job.

Comment (szabo137) [reply .gitlab-ci.yml]: Excellent! I like this solution.

**Review by szabo137 — APPROVED** [review]
All conversations are resolved, no additional concerns, ready for merging!

**Pull request merged**